ausroller/config.py

```diff
@@ -0,0 +1,119 @@
+        parser.add_argument('-s', '--secret', type=str, required=False,
+                            help='Path to file holding [<repopath>/secrets/<namespace>/secret_vars.json]')
+        args = parser.parse_args()
+
+        self.app_name = args.app
+        self.app_version = args.version
+        self.namespace = args.namespace
+        self.commit_message = args.message
+        self.is_dryrun = args.dryrun
+        self.is_dryrun_but_templates = args.dryruntemp
+        self.is_verbose = args.verbose
+        if self.is_verbose:
+            self.log_level=logging.DEBUG
+        else:
+            self.log_level=logging.INFO
+        self.configfile = args.config
+        self.extravarsfile = args.extravars
+        self.secretsfile = args.secret
+
+    def __init__(self):
+        # Default values
+        pass
+
+    def read_config(self):
+        home_dir = os.path.expanduser("~")
+        # read config file
+        if not self.configfile:
+            self.configfile = os.path.join(home_dir, ".ausroller.ini")
+        cp = ConfigParser.ConfigParser()
+        try:
+            cp.read(self.configfile)
+        except:
+            logging.error(
+                "Cannot read configuration file \"{}\"!".format(self.configfile))
+            sys.exit(1)
+
+        try:
+            self.repopath = os.path.realpath(cp.get('ausroller', 'repopath'))
+        except:
+            logging.error("Cannot read 'repopath' from configuration file \"{}\"!".format(
+                self.configfile))
+            sys.exit(1)
+
+        # set paths and read in the json file with the secrets
+        self.templates_path = os.path.join(self.repopath, 'templates')
+        self.rollout_path = os.path.join(
+            self.repopath, 'rollout', self.namespace)
+        if not self.secretsfile:
+            self.secretsfile = os.path.join(
+                self.repopath, 'secrets', self.namespace, 'secret_vars.json')
+        try:
+            self.variables = self.read_variables(self.secretsfile)
+        except KeyError as e:
+            logging.error("Cannot read secret variables from \"{}\"! [{}]".format(self.secretsfile, e))
+            sys.exit(1)
+
+        self.extra_variables = {}
+        if not self.extravarsfile:
+            default_extravarsfile = os.path.join(
+                self.repopath, 'manifests', self.namespace, 'extra_vars.json')
+            if os.path.exists(default_extravarsfile):
+                logging.info("found default extra vars file")
+                self.extravarsfile = default_extravarsfile
+
+        if self.extravarsfile:
+            try:
+                self.extra_variables = self.read_variables(self.extravarsfile)
+            except KeyError as e:
+                logging.error("Cannot read extra variables from \"{}\"! [{}]".format(self.extravarsfile, e))
+                sys.exit(1)
+        self.kubectl_cmd = 'kubectl --namespace={}'.format(self.namespace)
+
+    def _custom_json_pairs_hook(self, pairs):
```

Comment:
This is a helper function and does not use or change the instance. Therefore it should be at least a staticmethod.